Mathlib/Dynamics/Transitive.lean

```diff
@@ -0,0 +1,106 @@
+/-
+Copyright (c) 2025 Daniel Figueroa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Figueroa
+-/
+import Mathlib.Dynamics.Minimal
+
+/-!
+# Topologically transitive monoid actions
+
+In this file we define an action of a monoid `M` on a topological space `α` to be
+*topologically transitive* if for any pair of nonempty open sets `U` and `V` in `α` there exists an
+`m : M` such that `(m • U) ∩ V` is nonempty. We also provide an additive version of this definition
+and prove basic facts about topologically transitive actions.
+
+## Tags
+
+group action, topologically transitive
+-/
+
+
+open scoped Pointwise
+
+/-- An action of an additive monoid `M` on a topological space `α` is called
+*topologically transitive* if for any pair of nonempty open sets `U` and `V` in `α` there exists an
+`m : M` such that `(m +ᵥ U) ∩ V` is nonempty. -/
+class AddAction.IsTopologicallyTransitive (M α : Type*) [AddMonoid M] [TopologicalSpace α]
+    [AddAction M α] : Prop where
+  exists_vadd_inter : ∀ {U V : Set α}, IsOpen U → U.Nonempty → IsOpen V → V.Nonempty →
+    ∃ m : M, ((m +ᵥ U) ∩ V).Nonempty
+
+/-- An action of a monoid `M` on a topological space `α` is called *topologically transitive* if
+for any pair of nonempty open sets `U` and `V` in `α` there exists an `m : M` such that
+`(m • U) ∩ V` is nonempty. -/
+@[to_additive]
+class MulAction.IsTopologicallyTransitive (M α : Type*) [Monoid M] [TopologicalSpace α]
+    [MulAction M α] : Prop where
+  exists_smul_inter : ∀ {U V : Set α}, IsOpen U → U.Nonempty → IsOpen V → V.Nonempty →
+    ∃ m : M, ((m • U) ∩ V).Nonempty
+
+open MulAction Set
+
+variable (M : Type*) {α : Type*} [TopologicalSpace α] [Monoid M] [MulAction M α]
+
+section IsTopologicallyTransitive
+
+@[to_additive]
+theorem MulAction.isTopologicallyTransitive_iff :
+    IsTopologicallyTransitive M α ↔ ∀ {U V : Set α}, IsOpen U → U.Nonempty → IsOpen V →
+    V.Nonempty → ∃ m : M, ((m • U) ∩ V).Nonempty := ⟨fun h ↦ h.1, fun h ↦ ⟨h⟩⟩
+
+/-- A monoid action on `α` by `M` is topologically transitive if and only if for any nonempty
+open subset `U` of `α` the union over the elements of `M` of images of `U` is dense in `α`. -/
+@[to_additive]
+theorem MulAction.isTopologicallyTransitive_iff_dense_iUnion :
+    IsTopologicallyTransitive M α ↔
+    ∀ {U : Set α}, IsOpen U → U.Nonempty → Dense (⋃ m : M, m • U) := by
+  simp [isTopologicallyTransitive_iff, dense_iff_inter_open, inter_iUnion, inter_comm]
```

Comment:
Can you squeeze the simp using `simp?`. Otherwise, when we change the simp set, the output of `simp` might change and the next `exact` line would need fixing.